sound/soc/sof/intel/hda-dsp.c

```diff
@@ -406,6 +412,12 @@ static int hda_dsp_update_d0i3c_register(struct snd_sof_dev *sdev, u8 value)
 	}
 
 	reg = snd_sof_dsp_read8(sdev, HDA_DSP_HDA_BAR, chip->d0i3_offset);
+	/* Confirm d0i3 state is changed and report error in abnormal case */
+	if ((reg & SOF_HDA_VS_D0I3C_I3) != value) {
+		dev_err(sdev->dev, "failed to update D0I3C!\n");
+		return -EIO;
+	}
+
```

Comment:
commit title should be 

ASoC: SOF: Intel: hda-dsp: harden D0i3 programming sequence